index.js

```diff
@@ -58,11 +63,32 @@ function fastifyView (fastify, opts, next) {
       if (!that.res.getHeader('content-type')) {
         that.header('Content-Type', 'text/html')
       }
-
       that.send(lru.get(page)(data))
     }
   }
 
+  function viewEjsMate (page, data) {
+    if (!page || !data) {
+      this.send(new Error('Missing data'))
+      return
+    }
+    if (!options.settings) {
+      options.settings = {}
+    }
+    // ejs-mate use views to find layouts
+    options.settings.views = templatesDir
+    // setting locals to pass data by
+    options.locals = Object.assign({}, options.locals, data)
+    engine(join(templatesDir, page), options, send(this))
+
+    function send (that, data) {
+      return function _send (err, html) {
+        if (err) return that.send(err)
+        that.header('Content-Type', 'text/html').send(html)
+      }
+    }
```

Comment:
can you please move this closure-factory outside of this function? There is no need to allocate it here.

Reply:
sure. Good point.